micronota/db/tigrfam.py

```diff
@@ -0,0 +1,266 @@
+   TP  Always "TIGRFAMs", the identifier of this database
+   === ========================================================================
+
+Proteins that score
+above the trusted cutoffs are believed to reside within the family and
+those falling below the noise cutoffs are believed to reside outside the
+family.  The margin of error with respect to presence or absence of a
+protein within a TIGRFAMs family is represented by the score range
+between noise and trusted cutoffs.
+
+Because the number of completed and nearly completed genomes
+has now entered the thousands, protein families are becoming
+very large, and exceptions may be found in certain equivalog
+families.  Exceptions usually represent neofunctionalizations
+that arise within an equivalog family, although some may
+represent paralogs that are minimally derived since their
+branching from the equivalog familiy (i.e. short branch length).
+
+TIGRFAMs now includes 57 models of type "exception",  a type
+of model that overrules annotation from an equivalog model,
+either to give more specific information or to correct annotation
+for a neofunctionalized subgroup.
+
+TIGRFAMS_GO_LINK
+^^^^^^^^^^^^^^^^
+Gene Ontology (GO) term assignments.
+Models may have  0, 1, or several GO assignments.
+
+
+Reference
+---------
+.. [1] http://www.ncbi.nlm.nih.gov/pubmed/12520025
+.. [2] http://www.ncbi.nlm.nih.gov/pubmed/23197656
+
+'''
+
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import shutil
+import gzip
+import tarfile
+from tempfile import mkdtemp
+from os.path import join, exists
+from ftplib import FTP
+from sqlite3 import connect
+
+from ..bfillings.hmmer import hmmpress_hmm
+
+
+def prepare_db(out_d, prefix='tigrfam_v15.0', force=False,
+               server='ftp.tigr.org',
+               path='pub/data/TIGRFAMs',
+               hmm='TIGRFAMs_15.0_HMM.LIB.gz',
+               metadata='TIGRFAMs_15.0_INFO.tar.gz'):
+    '''Download and prepare TIGRFAM database.
+
+    Parameters
+    ----------
+    out_d : str
+        The directory of output files.
+    prefix : str
+        The file name (without extensions) of the output files.
+    force : boolean
+        Whether to overwrite existing files
+    server : str
+        FTP server to download TIGRFAM files
+    path : str
+        The path on FTP to find files to download
+    hmm : str
+        The file name of hmm models
+    metadata : str
+        The file name of the metadata for the hmm models
+    '''
+
+    hmm_out = join(out_d, '%s.hmm' % prefix)
+    if exists(hmm_out) and not force:
+        print('Database %s already exists. Skip it.' % prefix)
```

Comment:
Thanks for all the comments! much appreciated. will address

On Sun, Jan 31, 2016 at 8:40 PM, Yoshiki Vázquez Baeza <
notifications@github.com> wrote:

> In micronota/db/tigrfam.py
> <https://github.com/biocore/micronota/pull/40#discussion_r51378915>:
>
> > +        The file name (without extensions) of the output files.
> > +    force : boolean
> > +        Whether to overwrite existing files
> > +    server : str
> > +        FTP server to download TIGRFAM files
> > +    path : str
> > +        The path on FTP to find files to download
> > +    hmm : str
> > +        The file name of hmm models
> > +    metadata : str
> > +        The file name of the metadata for the hmm models
> > +    '''
> > +
> > +    hmm_out = join(out_d, '%s.hmm' % prefix)
> > +    if exists(hmm_out) and not force:
> > +        print('Database %s already exists. Skip it.' % prefix)
>
> Skip -> Skipping
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/40/files#r51378915>.
>